conf/nginx/nginx.conf.web.https.default.template

```diff
@@ -1,24 +1,26 @@
 # HTTPS Proxy Default Configuration
-#
-server {
-    ${core.ipboth.enabled}listen                  [::]:${web.https.port} default_server ipv6only=off;
-    ${core.ipv4only.enabled}listen                ${web.https.port} default_server;
-    ${core.ipv6only.enabled}listen                [::]:${web.https.port} default_server;
 
-    ssl                     on;
-    ssl_protocols           ${web.ssl.protocols};
-    ssl_prefer_server_ciphers ${web.ssl.preferserverciphers};
-    ssl_session_cache       ${ssl.session.cachesize};
-    ssl_session_timeout     ${ssl.session.timeout};
-    ssl_ciphers             ${web.ssl.ciphers};
-    ssl_ecdh_curve          ${web.ssl.ecdh.curve};
-    ssl_certificate         ${ssl.crt.default};
-    ssl_certificate_key     ${ssl.key.default};
-    ssl_verify_client       ${ssl.clientcertmode.default};
-    ssl_verify_depth        ${ssl.clientcertdepth.default};
-    ${web.ssl.dhparam.enabled}ssl_dhparam             ${web.ssl.dhparam.file};
-    return 444;
-}
+${web.strict.servername}# Strict servername enforcing block
+${web.strict.servername}# Enabled/disabled through the 'zimbraReverseProxyStrictServerName' configuration item
+${web.strict.servername}server {
+${web.strict.servername}    ${core.ipboth.enabled}listen                  [::]:${web.https.port} default_server ipv6only=off;
+${web.strict.servername}    ${core.ipv4only.enabled}listen                ${web.https.port} default_server;
+${web.strict.servername}    ${core.ipv6only.enabled}listen                [::]:${web.https.port} default_server;
+${web.strict.servername}
+${web.strict.servername}    ssl                     on;
+${web.strict.servername}    ssl_protocols           ${web.ssl.protocols};
+${web.strict.servername}    ssl_prefer_server_ciphers ${web.ssl.preferserverciphers};
+${web.strict.servername}    ssl_session_cache       ${ssl.session.cachesize};
+${web.strict.servername}    ssl_session_timeout     ${ssl.session.timeout};
+${web.strict.servername}    ssl_ciphers             ${web.ssl.ciphers};
+${web.strict.servername}    ssl_ecdh_curve          ${web.ssl.ecdh.curve};
+${web.strict.servername}    ssl_certificate         ${ssl.crt.default};
+${web.strict.servername}    ssl_certificate_key     ${ssl.key.default};
+${web.strict.servername}    ssl_verify_client       ${ssl.clientcertmode.default};
+${web.strict.servername}    ssl_verify_depth        ${ssl.clientcertdepth.default};
+${web.strict.servername}    ${web.ssl.dhparam.enabled}ssl_dhparam             ${web.ssl.dhparam.file};
+${web.strict.servername}    return 400;
```

Comment:
I think this should be the only line which is conditional via `${web.strict.servername}`. Ie. the original `server` block (the one from before ) with the `default_server` should stay. Rationale: Without such a `server` block nginx will pick the first one instead (cf. https://nginx.org/en/docs/http/request_processing.html) which would make the whole behaviour a bit less deterministic.

There should also be a line

    server_name _;

which is the idiomatic way of specifying the default block (cf. https://nginx.org/en/docs/http/server_names.html#miscellaneous_names). Without this line I'd expect the block to match only request *without* any `Host` header at all (at least if I interpret the page linked above correctly).

Reply:
If the 'server_name _' is missing it is implied.  i.e. it's the same if it's present or not.

Reply:
If ONLY the 'return 400' line is conditional what is to be returned otherwise?
This will only match in the case that there isn't anything more specific.
NOTE: An IP based server is more specific than this one.

Reply:
FYI: 'server_name _' was added to the section